src/_pytest/timing.py

```diff
@@ -7,6 +7,47 @@
 """
 from time import perf_counter
 from time import sleep
-from time import time
+from time import time as Time
 
-__all__ = ["perf_counter", "sleep", "time"]
+
+_paused_for = 0.0
+_pause_start = 0.0
+
+
+def stop() -> None:
```

Comment:
i beleive based on the initial input, if we add this, this should be a class and be used in a fixture

Reply:
Why should this be a class ?

I tested it with a local install on my original project and it worked great...

Reply:
@RonnyPfannschmidt did you see the docs I added?
I showed there how I think this mechanism should be used.

Reply:
Originally I was inspired by - [T.StartTimer](https://golang.org/pkg/testing/#B.StartTimer) & [T.StopTimer](https://golang.org/pkg/testing/#B.StopTimer) from Golang's testing library:

> StopTimer stops timing a test. This can be used to pause the timer
> while performing complex initialization that you don't
> want to measure.

From: https://golang.org/src/testing/benchmark.go?s=3884:3907#L124


Though the StartTimer I implemented partially differs because in pytest the time all the tests took is measured by a simple difference between start time and "now" (which is implemented using `pytest.timing.time()`), so I approached this by subtracting the "pause" time from the `pytest.timing.time()` function.

Originally in Golang (https://golang.org/src/testing/benchmark.go?s=3538:3562#L111):

> StartTimer starts timing a test. This function is called automatically
> before a benchmark starts, but it can also be used to resume timing after
> a call to StopTimer.